### PR TITLE
outline for cloudant query/search

### PR DESCRIPTION
**Pull request by jcounts**
@evansde77 @petevg @skscharr 

Just getting a basic outline of a few things needed to implement Cloudant query.

### CODE REVIEW

**Comment by evansde77** [comment]
Tap in @alfinkel on this as well, he has been thinking about search and query APIs too

File: src/cloudant/database.py [hunk 1]
```diff
@@ -489,6 +489,32 @@ def unshare_database(self, username):
         resp.raise_for_status()
         return resp.json()
 
+    def index(self):
```
Comment (evansde77) [inline src/cloudant/database.py:492]: Better call Kenny Loggins, cause your in the Danger Zone... 
This clobbers self.index in the base class, right? search_indices maybe?

File: src/cloudant/database.py [hunk 2]
```diff
@@ -489,6 +489,32 @@ def unshare_database(self, username):
...
+        """
+        GET /db/_index Returns existing indexes
+        """
+        pass
+
+    def find(self, selector, **kwargs):
+        """
+        POST /db/_find
```
Comment (pengale) [inline src/cloudant/database.py:500]: Is it really a POST on the other end? That's sad ... a search should really be a GET.

**Comment by jcounts** [comment]
Closing out as this repo is no longer active. See: https://github.com/cloudant/python-cloudant

**Pull request closed without merging**